service/history/workflow/task_refresher.go

```diff
@@ -63,12 +71,29 @@ func NewTaskRefresher(
 	}
 }
 
-func (r *TaskRefresherImpl) RefreshTasks(
+func (r *TaskRefresherImpl) Refresh(
 	ctx context.Context,
 	mutableState MutableState,
 ) error {
 	// Invalidate all tasks generated for this mutable state before the refresh.
 	mutableState.GetExecutionInfo().TaskGenerationShardClockTimestamp = r.shard.CurrentVectorClock().GetClock()
+	return r.PartialRefresh(ctx, mutableState, nil)
+}
+
+func (r *TaskRefresherImpl) PartialRefresh(
+	ctx context.Context,
+	mutableState MutableState,
+	minVersionedTransition *persistencespb.VersionedTransition,
+) error {
+
+	if minVersionedTransition != nil {
+		if err := TransitionHistoryStalenessCheck(
```

Comment:
Can you please put a comment here to explain why this is needed. It took me a second to process and I want devs without context to understand this process better.